internal/model/netx.go

```diff
@@ -303,6 +303,26 @@ type Trace interface {
 	// can use functionality exported by the ./internal/testingx pkg.
 	TimeNow() time.Time
 
+	// OnDNSLookupDone is called when DNSLookup terminates
```

Comment:
Beware that the function name has changed below. I think you should also clarify that this gets called when you are using a transport as opposed to another function that we'll need for getaddrinfo.

I think the very important take home message here is that this callback is called after a single round trip.